### DISCUSSION TITLE
Adding footnotes to pages (layout: page)

### DISCUSSION THREAD

**Discussion by liberomureddu** [alshedivat/al-folio#2687] [Q&A]
I realised that the only way to have footnotes is to use the distill layout. I tried my best to make a project page to show footnotes but without success. In some of my projects, as they are connected to my research, it'd be nice to have some footnotes and, why note, references. 
The reason to not use the distill layout is that it looks different from the page layout. Moreover, I wouldn't like my projects to appear as blog posts, with date, etc.
Any hint? Thanks in advance.

**Comment by george-gca** [comment]
Bibliography is [already supported](https://github.com/alshedivat/al-folio/blob/97f78e5fb883a4bedd0c1e175ce69daadd4a876f/_layouts/page.liquid#L14), check the [example](https://alshedivat.github.io/al-folio/projects/1_project/). To enable this, just add `related_publications: true` to the project front matter.

About footnotes in the distill style, this is currently not supported, but could be implemented. The footnotes are handled by [assets/js/distillpub/template.v2.js](https://github.com/alshedivat/al-folio/blob/master/assets/js/distillpub/template.v2.js). If you want to do it in the same style, you'll need to understand how it handles it and recreate the same behavior in the `page` layout.

**Reply by liberomureddu** [reply]
Great to know about bibliography, I somehow missed that! I'll investigate how to add footnotes to pages.

**Reply by liberomureddu** [reply]
Hi, I think I found a way to make it work. Do you think it's a good approach? A minor issue now is the layout of the footnotes, which is very different from the references, working on it. 
- In the file _layouts/default.liquid I added 
`<script src="{{ '/assets/js/distillpub/template.v2.js' | relative_url }}"></script>`
to the `<head>` section
- in the same file I added
 ```
{% if page.footnotes == true %}
     <d-appendix>
        <d-footnote-list></d-footnote-list>
        <d-citation-list></d-citation-list>
      </d-appendix>
    {% endif %}
```
so that I can add in a project page:
`footnotes: true` when I want to have footnotes displayed.

**Reply by george-gca** [reply]
By adding `<script src="{{ '/assets/js/distillpub/template.v2.js' | relative_url }}"></script>` you are adding almost the whole distill script, and as such its behavior and looks. This will lead to a lot of unexpected behavior.

The best solution would be for you to understand how distill script handles the footnotes (meaning analyzing the script, understanding it) and recreate the same behavior in the page layout.

**Reply by liberomureddu** [reply]
Yes, I just noticed that it's changed the appearance of some elements. Thanks for the tip, I'll invest some time analysing the script. Thanks for the feedback!
